src/ddprof.cc

```diff
@@ -118,7 +118,11 @@ DDRes ddprof_teardown(DDProfContext *ctx) {
     LG_WRN("Error when calling pevent_cleanup.");
   }
 
-  DDRES_CHECK_FWD(ddprof_stats_free());
+  if (IsDDResNotOK(ddprof_stats_free())) {
+    LG_WRN("Error when calling ddprof_stats_free.");
+  }
+
+  LOG_close();
```

Comment:
minor: this is better, though not symmetrical the setup is done in `ddprof_context_setup`
should we not consider the close in `ddprof_context_free` ? 
This is still not optimal as the logger is global (though it could be part of the ddprof context)

Reply:
Nice suggestion, I will to the change !